atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2123,6 +2124,65 @@ public void compactInternally(TableReference tableRef) {
         }
     }
 
+    @Override
+    public NodeAvailabilityStatus getNodeAvailabilityStatus() {
+        if (!doesConfigReplicationFactorMatchWithCluster()) {
+            return NodeAvailabilityStatus.TERMINAL;
+        }
+        return getStatusByRunningOperationsOnEachHost();
```

Comment:
I think the ordering is backwards here. You're checking that keyspace config is correct before you're even sure that the cluster is up. What happens if all nodes are down? Will `doesConfigReplicationFactorMatchWithCluster()` return true because it can't tell or false because the operation throws an exception because it can't reach the node.